xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -685,9 +685,19 @@ bool CEdl::AddCut(Cut& cut)
     int autowind = g_advancedSettings.m_iEdlCommBreakAutowind * 1000; // seconds -> ms
 
     if (cut.start > 0) // Only autowait if not at the start.
-     cut.start += autowait;
-    if (cut.end > cut.start + autowind) // Only autowind if it won't go back past the start (should never happen).
-     cut.end -= autowind;
+    {
+      /* get the cut length so we don't start skipping after the end */
+      int cutLength = cut.end - cut.start;
+      /* add the lesser of the cut length or the autowait to the start */
+      cut.start += autowait > cutLength ? cutLength : autowait;
+    }
+    if (cut.end > cut.start) // Only autowind if there is any cut time remaining.
+    {
+      /* get the remaining cut length so we don't rewind to before the start */
+      int cutLength = cut.end - cut.start;
+      /* subtract the lesser of the cut length or the autowind from the end */
+      cut.end -= autowind > cutLength ? cutLength : autowind;
+    }
```

Comment:
> I'm hesitant to comment because of the don't comment rule

@DerickLebron what do you mean? this is your PR, you can comment as much as you want